### ISSUE TITLE
Performance improvement

### ISSUE THREAD

**Issue by tamada** [tamada/uniq2#8]
The current implementation does not optimize runtime performance.
Therefore, it is quite slow (see #5, reported by @sergeevabc).

`uniq2` should improve runtime performance.

Thank you @sergeevabc

**Comment by tamada** [comment]
resolved at #9 

change algorithm, and structure of `uniq2`.

@sergeevabc
try version 1.1.1
https://github.com/tamada/uniq2/releases/tag/v1.1.1

```
$ time uniq2 hostnames.txt > /dev/null
real	0m0.050s
user	0m0.046s
sys	0m0.009s

$ time awk 'NF && !line[$0]++' hostnames.txt > /dev/null
real	0m0.145s
user	0m0.138s
sys	0m0.006s
```

**Comment by sergeevabc** [comment]
Indeed, the performance has been improved. Well done, @tamada.

```console
$ procprofile -u -- uniq2 hostnames.txt out_uniq2.txt

Process ID       : 1932
Thread ID        : 5384
Process Exit Code: 0
Thread Exit Code : 0

User Time        : 0.046s
Kernel Time      : 0.031s
Process Time     : 0.077s
Clock Time       : 0.108s

Working Set      : 14328 KB
Paged Pool       : 98 KB
Nonpaged Pool    : 10 KB
Pagefile         : 23716 KB
Page Fault Count : 3623

IO Read          : 1212 KB (in 305 reads)
IO Write         : 1184 KB (in 297 writes)
IO Other         : 1 KB (in 85 others)
```
